kolibri/core/views.py

```diff
@@ -38,3 +47,47 @@ def set_language(request):
                                 path=settings.LANGUAGE_COOKIE_PATH,
                                 domain=settings.LANGUAGE_COOKIE_DOMAIN)
     return response
+
+
+def logout_view(request):
+    logout(request)
+    return http.HttpResponseRedirect('/')
+
+
+def get_url_by_role(role, first_login):
+    obj = next((hook for hook in RoleBasedRedirectHook().registered_hooks
+                if hook.role == role and hook.first_login == first_login), None)
+    if obj:
+        return obj.url
+
+
+class GuestRedirectView(View):
+    def get(self, request):
+        """
+        Redirects a guest user to a learner accessible page.
+        """
+        return HttpResponseRedirect(get_url_by_role(user_kinds.LEARNER, False))
+
+
+class RootURLRedirectView(View):
+
+    def get(self, request):
+        """
+        Redirects user based on the highest role they have for which a redirect is defined.
+        """
+        first_login = request.session.get("first_login", False)
+        if request.user.is_authenticated():
+            url = None
+            if request.user.is_superuser:
+                url = url or get_url_by_role(user_kinds.SUPERUSER, first_login)
+            roles = set(Role.objects.filter(user_id=request.user.id).values_list('kind', flat=True).distinct())
+            if user_kinds.ADMIN in roles:
```

Comment:
Use else if statements instead?

Reply:
I thought about that, but it doesn't quite do it, because you need to be able to fall through from an admin user to a coach user to a learner (so if nothing is defined for an admin, they still get redirected to wherever a coach or a learner would end up).